src/preferences/preferences.cpp

```diff
@@ -1550,6 +1550,35 @@ const std::vector<game_config::server_info>& prefs::builtin_servers_list()
 	return pref_servers;
 }
 
+void prefs::add_game_preset(config&& preset_data)
+{
+	config preset{ prefs_list::game_preset, std::move(preset_data) };
+
+	if(preset.has_child(prefs_list::game_preset) && preset.all_children_count() == 1 && preset.attribute_count() == 0) {
```

Comment:
This condition block is not needed anymore, since you're creating a valid tag structure here in the function. Just drop it.